### PR TITLE
using bool instead of error

### PR DESCRIPTION
**Pull request by zouyee**
**What this PR does / why we need it**:

CommonAccessor returns a Common interface for the provided object or a boolean value wether the object to provide list

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:

**Special notes for your reviewer**:

**Release note**:

```NONE

```

### CODE REVIEW

File: pkg/api/ref/ref.go [hunk 1]
```diff
@@ -65,8 +65,8 @@ func GetReference(scheme *runtime.Scheme, obj runtime.Object) (*api.ObjectRefere
 	var listMeta metav1.Common
 	objectMeta, err := meta.Accessor(obj)
 	if err != nil {
-		listMeta, err = meta.CommonAccessor(obj)
-		if err != nil {
+		listMeta, boolean = meta.CommonAccessor(obj)
```
Comment (enj) [inline pkg/api/ref/ref.go]: The convention is `foo, ok = ...`

File: staging/src/k8s.io/apiserver/pkg/registry/rest/table.go [hunk 2]
```diff
@@ -63,12 +63,12 @@ func (c defaultTableConvertor) ConvertToTable(ctx genericapirequest.Context, obj
...
 		}
 	}
-	if m, err := meta.ListAccessor(object); err == nil {
+	if m, boolean := meta.ListAccessor(object); !boolean {
 		table.ResourceVersion = m.GetResourceVersion()
 		table.SelfLink = m.GetSelfLink()
 		table.Continue = m.GetContinue()
 	} else {
-		if m, err := meta.CommonAccessor(object); err == nil {
+		if m, boolean := meta.CommonAccessor(object); err == nil {
```
Comment (enj) [inline staging/src/k8s.io/apiserver/pkg/registry/rest/table.go]: Bug: `err == nil` is not valid, should be checking the bool.

File: staging/src/k8s.io/apiserver/pkg/registry/rest/table.go [hunk 3]
```diff
@@ -63,12 +63,12 @@ func (c defaultTableConvertor) ConvertToTable(ctx genericapirequest.Context, obj
 			return nil, err
 		}
 	}
-	if m, err := meta.ListAccessor(object); err == nil {
+	if m, boolean := meta.ListAccessor(object); !boolean {
```
Comment (enj) [inline staging/src/k8s.io/apiserver/pkg/registry/rest/table.go]: Check is inverted.

File: staging/src/k8s.io/apimachinery/pkg/api/meta/meta.go [hunk 4]
```diff
@@ -36,34 +36,33 @@ var errNotList = fmt.Errorf("object does not implement the List interfaces")
 
 var errNotCommon = fmt.Errorf("object does not implement the common interface for accessing the SelfLink")
 
-// CommonAccessor returns a Common interface for the provided object or an error if the object does
-// not provide List.
-// TODO: return bool instead of error
-func CommonAccessor(obj interface{}) (metav1.Common, error) {
+// CommonAccessor returns a Common interface for the provided object or a boolean value wether the object to
```
Comment (enj) [inline staging/src/k8s.io/apimachinery/pkg/api/meta/meta.go]:
```
CommonAccessor returns the metav1.Common interface for the provided object.
The bool must be used to determine if the object provides this interface.
```

**Review by enj — CHANGES_REQUESTED** [review]

**Comment by enj** [comment]
@sttts @smarterclayton are we trying to prevent changes to the staging bits?

**Comment by sttts** [comment]
> are we trying to prevent changes to the staging bits?

No, we don't. I don't expect many 3rdparty projects which use this functionality (none visible at https://github.com/search?p=1&q=CommonAccessor+language%3AGo&type=Code&utf8=%E2%9C%93). Moreover, the fix is obvious and golang complains.

**Pull request closed without merging**